chore: Follow-on PR to fully enable onheap memory usage (#1210)
apache/datafusion-comet: docs/source/user-guide/tuning.md

## Memory Tuning

### Unified Memory Management with Off-Heap Memory

The recommended way to share memory between Spark and Comet is to set `spark.memory.offHeap.enabled=true`. This allows

> Comet to share an off-heap memory pool with Spark. The size of the pool is specified by `spark.memory.offHeap.size`.

Reviewer: ```suggestion
Comet to share an off-heap memory pool with Spark. The size of the pool is specified by `spark.memory.offHeap.size`. For more details about Spark off-heap memory mode, please refer to Spark documentation: https://spark.apache.org/docs/latest/configuration.html.
```

### Dedicated Comet Memory Pools

> If the `spark.memory.offHeap.enabled` setting is not enabled then Comet will use its own dedicated memory pools that

Reviewer: ```suggestion
Spark uses on-heap memory mode by default, i.e., the `spark.memory.offHeap.enabled` setting is not enabled. If Spark is under on-heap memory mode, Comet will use its own dedicated memory pools that
```

are not shared with Spark. This requires additional configuration settings to be specified to set the size and type of
memory pool to use.

The size of the pool can be set explicitly with `spark.comet.memoryOverhead`. If this setting is not specified then
the memory overhead will be calculated by multiplying the executor memory by `spark.comet.memory.overhead.factor`
(defaults to `0.2`).

The type of pool can be specified with `spark.comet.exec.memoryPool`. The default setting is `greedy_task_shared`.

The valid pool types are:

- `greedy`
- `greedy_global`
- `greedy_task_shared`
- `fair_spill`
- `fair_spill_global`
- `fair_spill_task_shared`

> Pool types ending with `_global` use a single global memory pool between all tasks.

Reviewer: ```suggestion
Pool types ending with `_global` use a single global memory pool between all tasks on same executor.
```

Pool types ending with `_task_shared` share a single memory pool across all attempts for a single task.

> Other pool types create a dedicated pool per task using a fraction of the available pool size based on number of cores 
> and cores per task.

Reviewer 2: ```suggestion
Other pool types create a dedicated pool per native query plan using a fraction of the available pool size based on number of cores 
and cores per task.
```
